src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java

```diff
@@ -1518,22 +1525,32 @@ private void visitJavaDoc(JavadocComment javadoc) {
         addToken(new Token(DOCUMENTATION_RANGE_END));
     }
 
-    private void indent() {
-        indent += 4;
-    }
-
-    private void unindent() {
-        indent = Math.max(indent - 4, 0);
+    private void indentedBlock(Runnable indentedSection) {
+        indentLevel++;
+        indentedSection.run();
+        indentLevel--;
     }
 
     private Token makeWhitespace() {
         // Use a byte array with Arrays.fill with empty space (' ') character rather than StringBuilder as StringBuilder
         // will check that it has sufficient size every time a new character is appended. We know ahead of time the size
         // needed and can remove all those checks by removing usage of StringBuilder with this simpler pattern.
-        byte[] bytes = new byte[indent];
+        return new Token(WHITESPACE, makeWhitespace(indentLevel));
+    }
+
+    private static String makeWhitespace(int indentLevel) {
+        if (indentLevel < INDENT_LEVELS.size()) {
+            return INDENT_LEVELS.get(indentLevel);
+        }
+
+        return makeNewWhitespace(indentLevel * 4);
```

Comment:
Should we add the new indent level to the cached list (or use a map instead) to avoid recreating it again?

Reply:
I'm not expecting for there to be indentation levels this deep and if there are they can be created on use